sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/XmlOptions.scala

```diff
@@ -66,7 +66,11 @@ private[sql] class XmlOptions(
 
   val compressionCodec = parameters.get(COMPRESSION).map(CompressionCodecs.getCodecClassName)
   val rowTagOpt = parameters.get(XmlOptions.ROW_TAG).map(_.trim)
-  require(!rowTagRequired || rowTagOpt.isDefined, s"'${XmlOptions.ROW_TAG}' option is required.")
+
+  if (rowTagRequired && rowTagOpt.isEmpty) {
+    throw QueryCompilationErrors.xmlRowTagRequiredError(XmlOptions.ROW_TAG)
```

Comment:
Because `val rowTag = rowTagOpt.getOrElse(XmlOptions.DEFAULT_ROW_TAG)`, do we really need throw the error?

Reply:
As discussed in the earlier [PR](https://github.com/apache/spark/pull/43389), the default row tag is not going to match any real XML files and confuse users. This PR standardize this error for other services built on top of this.

Reply:
if so, we should change `val rowTag = rowTagOpt.getOrElse(XmlOptions.DEFAULT_ROW_TAG)` to `val rowTag = rowTagOpt.get`

Reply:
That would make `rowTag` required for other paths like `to_xml`, where it is ok to use the default.

Reply:
So, the error appears if users call `to_xml` and do not specify the rowTag option?

Reply:
Currently, `to_xml` doesn't throw an error if `rowTag` is not specified because it uses the default. If we do the change that you suggested above, even `to_xml` will require `rowTag` and throw an error if one is not provided. This PR is just standardizing the error that gets thrown when rowTag is not provided on "read". It doesn't change any other behavior.

Reply:
I got it now. I'm missing the `rowTagRequired`. The `rowTagRequired` is false if the call the `to_xml`.